File: frontend/src/api/mcpService.ts
import request from '@/utils/request'

export interface McpService {
  id: number
  name: string
  code: string
  description: string
  enabled: number
  configUrl: string
  configAuthType: string
  configCredentials: string
  configTimeout: number
  configMethod: string
  sort: number
  command: string
  args: string
  env: string
  createdTime?: string
  updatedTime?: string
}

export interface CreateMcpServiceRequest {
  name: string
  code: string
  description?: string
  enabled?: number
  configUrl?: string
  configAuthType?: string
  configCredentials?: string
  configTimeout?: number
  configMethod?: string
  sort?: number
  command?: string
  args?: string
  env?: string
}

export function getMcpServices() {
  return request.get<McpService[]>('/v1/mcp/services')
}

export function getMcpService(id: number) {
  return request.get<McpService>(`/v1/mcp/services/${id}`)
}

export function createMcpService(data: CreateMcpServiceRequest) {
  return request.post<number>('/v1/mcp/services', data)
}

export function updateMcpService(id: number, data: CreateMcpServiceRequest) {
  return request.put(`/v1/mcp/services/${id}`, data)
}

export function deleteMcpService(id: number) {
  return request.delete(`/v1/mcp/services/${id}`)
}

export function toggleMcpService(id: number) {
  return request.put(`/v1/mcp/services/${id}/toggle`)
}

File: backend/src/main/java/com/police/kb/domain/entity/McpService.java
package com.police.kb.domain.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * MCP 服务实体类
 */
@Data
@TableName("mcp_service")
public class McpService {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String code;

    private String description;

    private Integer enabled;

    @TableField("config_url")
    private String configUrl;

    @TableField("config_auth_type")
    private String configAuthType;

    @TableField("config_credentials")
    private String configCredentials;

    @TableField("config_timeout")
    private Integer configTimeout;

    @TableField("config_method")
    private String configMethod;

    private Integer sort;

    private String command;

    private String args;

    private String env;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdTime;

    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    @TableLogic
    private Integer deleted;
}
